src/NATS.Client.ObjectStore/NatsObjStore.cs

```diff
@@ -542,6 +542,10 @@ public async ValueTask<ObjectMetadata> GetInfoAsync(string key, bool showDeleted
             var base64String = Convert.FromBase64String(response.Message.Data);
             var data = NatsObjJsonSerializer<ObjectMetadata>.Default.Deserialize(new ReadOnlySequence<byte>(base64String)) ?? throw new NatsObjException("Can't deserialize object metadata");
 
+            if (!DateTimeOffset.TryParse(response.Message.Time, out var created))
+                throw new NatsObjException("Can't parse timestamp message value");
+            data.MTime = created;
+
```

Comment:
This is good. matches the Go implementation: [implementation](https://github.com/nats-io/nats.go/blob/52a2d33b791b6e819aacdaeef2c0b79c2cfbf81b/object.go#L975), [type definition](https://github.com/nats-io/nats.go/blob/833574b050e80aa552e3f78e9767251159bb3452/jsm.go#L1205)

...except rather than parsing here we should change the `Time`'s type to `DateTimeOffset` instead here:

https://github.com/nats-io/nats.net/blob/b09b9ae09aa0751677690091fd066029e28b98e0/src/NATS.Client.JetStream/Models/StoredMessage.cs#L34-L46